plugins/woocommerce/src/Internal/Admin/WCPayPromotion/Init.php

```diff
@@ -19,7 +19,17 @@ class Init extends RemoteSpecsEngine {
 	 * Constructor.
 	 */
 	public function __construct() {
-		$is_payments_page = isset( $_GET['page'] ) && $_GET['page'] === 'wc-settings' && isset( $_GET['tab'] ) && $_GET['tab'] === 'checkout'; // phpcs:ignore WordPress.Security.NonceVerification
+		/* phpcs:disable WordPress.Security.NonceVerification */
+		$is_payments_page   = isset( $_GET['page'] ) && 'wc-settings' === $_GET['page'] && isset( $_GET['tab'] ) && 'checkout' === $_GET['tab'];
+		$is_wc_admin_page   = isset( $_GET['page'] ) && 'wc-admin' === $_GET['page'];
+		$is_payments_task   = $is_wc_admin_page && isset( $_GET['task'] ) && 'payments' === $_GET['task'];
+		$is_payment_welcome = $is_wc_admin_page && isset( $_GET['path'] ) && '/wc-pay-welcome-page' === $_GET['path'];
+		/* phpcs:enable */
+
+		if ( $is_payments_page || $is_payments_task || $is_payment_welcome ) {
+			add_filter( 'woocommerce_admin_shared_settings', array( $this, 'add_component_settings' ) );
```

Comment:
I think this isn't reliable, partially because it's not reachable via react navigation:

1. Set store country to `United States - Alabama`
2. Go to Homescreen
3. Click on payments task
4. Observe the screen doesn't show `WooPay`
5. Refresh
6. Observe the screen show `WooPay`

#### Navigated via react:

![TJ8OnP.png](https://github.com/user-attachments/assets/1329c7ce-b9e6-4d9b-bac7-572dc3241e6a)

#### Full refresh:


![TaDlFe.png](https://github.com/user-attachments/assets/b773af1f-0bbd-4f34-bbf9-8f07003ac3a4)

Reply:
Thanks, good catch! I'll update the PR to load `isWooPayEligible` for all WC admin pages.

Reply:
Fixed in a4a9449479aa2f94773c1cf8abb82459bd41c1bb.